source/constructs/api/data_source/glue_database_detector.py

```diff
@@ -0,0 +1,125 @@
+import os
+
+import boto3
+import logging
+
+from common.constant import const
+from common.enum import ConnectionState, DatabaseType, Provider
+from db.database import get_session
+from db.models_data_source import DetectionHistory, RdsInstanceSource, Account
+from . import crud
+from . import service
+from catalog.service import delete_catalog_by_database_region
+from sqlalchemy.orm import Session
+import asyncio
+
+sts_client = boto3.client('sts')
+admin_account_region = boto3.session.Session().region_name
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+
+async def detect_glue_database_connection(session: Session, aws_account_id: str):
+    iam_role_name = crud.get_iam_role(aws_account_id)
+    history = DetectionHistory(aws_account=aws_account_id, source_type='jdbc', state=0)
+    session.add(history)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem:**
 This line of code lacks validation when processing input data through the following parameter: &#39;session&#39; (index: 0 | type: Session). The parameter is exposed to external callers, because its enclosing class and method are publicly accessible. This means that upstream validation, if it exists, can be bypassed. Other validated parameters: &#39;aws_account_id&#39;. Malicious, malformed, or unbounded inputs can cause unexpected runtime behavior or crashes, and can slow performance.

**Fix:**
 Add checks to ensure the validity of the parameter&#39;s value, such as testing it for nullness, emptiness, or equality. Or to prevent direct calls to it, [reduce the method&#39;s visibility](https://www.python.org/dev/peps/pep-0008/#descriptive-naming-styles) using single or double underscore.

 Learn more about potential threats and guidance from the [Common Weakness Enumeration website](https://cwe.mitre.org/data/definitions/693.html) and the [OWASP Cheat Sheet series](https://cheatsheetseries.owasp.org/cheatsheets/Input_Validation_Cheat_Sheet.html).